assets/src/editor/events/dates-and-times/editor-date/grid-view/event-date-details-panel.js

```diff
@@ -7,61 +7,66 @@ import { parseInfinity } from '@eventespresso/utils';
 import { __ } from '@eventespresso/i18n';
 import { InfinitySymbol } from '@eventespresso/value-objects';
 import PropTypes from 'prop-types';
+import { useEventDateUpdateRelatedTickets } from '@eventespresso/hooks';
 
 /**
  * Internal dependencies
  */
 import EventDateRegistrationsLink from '../event-date-registrations-link';
 
-const EventDateDetailsPanel = ( { eventDate } ) => useMemo(
-	() => {
-		const details = [
-			{
-				id: 'ee-event-date-sold',
-				label: __( 'sold', 'event_espresso' ),
-				value: eventDate.sold || 0,
-			},
-			{
-				id: 'ee-event-date-reserved',
-				label: __( 'reserved', 'event_espresso' ),
-				value: eventDate.reserved || 0,
-			},
-			{
-				id: 'ee-event-date-capacity',
-				label: __( 'capacity', 'event_espresso' ),
-				value: <InfinitySymbol value={ eventDate.regLimit } asInt />,
-				editable: {
-					type: 'text',
-					valueType: 'infinite',
-					onChange: ( cap ) => {
-						cap = parseInfinity( cap, true, true );
-						eventDate.regLimit = cap;
-						return <InfinitySymbol value={ cap } asInt />;
+const EventDateDetailsPanel = ( { eventDate } ) => {
+	const updateRelatedTickets = useEventDateUpdateRelatedTickets( eventDate );
+	return useMemo(
+		() => {
+			const details = [
+				{
+					id: 'ee-event-date-sold',
+					label: __( 'sold', 'event_espresso' ),
+					value: eventDate.sold || 0,
+				},
+				{
+					id: 'ee-event-date-reserved',
+					label: __( 'reserved', 'event_espresso' ),
+					value: eventDate.reserved || 0,
+				},
+				{
+					id: 'ee-event-date-capacity',
+					label: __( 'capacity', 'event_espresso' ),
+					value: <InfinitySymbol value={ eventDate.regLimit } asInt />,
+					editable: {
+						type: 'text',
+						valueType: 'infinite',
+						onChange: ( cap ) => {
+							cap = parseInfinity( cap, true, true );
+							eventDate.regLimit = cap;
+							updateRelatedTickets( { capacity: cap } );
+							return <InfinitySymbol value={ cap } asInt />;
```

Comment:
What about:
```
const capacity = parseInfinity( cap, true, true );
eventDate.regLimit = cap;
updateRelatedTickets( { capacity } );
return <InfinitySymbol value={ cap } asInt />;
```
this way we don't mutate cap

Reply:
Yes it makes the code better. Thanks

Reply:
Done :)

Reply:
Thanks :bowing_man: